src/class.c

```diff
@@ -1205,6 +1205,17 @@ void ClassDeclaration::addLocalClass(ClassDeclarations *aclasses)
     aclasses->push(this);
 }
 
+/***************************************
+ * Return true if class is an InputRange.
+ * This is defined as having the members: empty, front, popFront.
+ */
+bool ClassDeclaration::isInputRange()
+{
+    return (this->search(Loc(), Id::Ffront) &&
+            this->search(Loc(), Id::FpopFront) &&
+            this->search(Loc(), Id::Fempty));
+}
+
```

Comment:
There is probably other diagnostics that may find other range predicates useful. But why not add them all at once in this PR?